dev-java/hamcrest/hamcrest-2.2.ebuild

```diff
@@ -28,9 +28,9 @@ RDEPEND="
 	>=virtual/jre-1.8:*
 "
 
-# With jdk-11 or -17 there are compilation errors.
-# Some of them could be patched but for these 4 errors could not find a solution.
-# hamcrest/src/main/java/org/hamcrest/core/AnyOf.java:49: error: cannot find symbol
+# After adding "import java.lang.Iterable;"
+
+# hamcrest/src/main/java/org/hamcrest/core/AnyOf.java:50: error: cannot find symbol
 #         return anyOf((Iterable<Matcher<? super T>>) Arrays.asIterable(matchers));
```

Comment:
i searched java doc for `Arrays.asIterable()` but couldn't find it anywhere... the fix might be just to change it to `Arrays.asList()`.

Reply:
> i searched java doc for `Arrays.asIterable()` but couldn't find it anywhere... the fix might be just to change it to `Arrays.asList()`.

@fordfrog That clearly solves 2 of 4 errors.  It remains 2 errors with [CombinableMatcher.java](https://github.com/gentoo/gentoo/pull/26078/files#diff-716f5c311f82a39286c78c6a5778c86834d7feb91af5d914d9cbc69499230ff9R31-R53)

Reply:
here's the updated patch: [hamcrest-2.2-java-11.patch](https://github.com/gentoo/gentoo/files/9034205/hamcrest-2.2-java-11.patch.txt)
with the updated patch it now compiles for me even with `openjdk-17`.